### PR TITLE
Feature: Auto-login in dev mode

### PR DESCRIPTION
**Pull request by AntonNazarenko**
📨 ASANA ticket: https://app.asana.com/0/1125859137800433/1200061962500088/f

📝 PR Description:
This PR adds the auto-login ability to any user who runs the app in a browser.
It can be enabled and disabled either on the login screen or on the settings page.
Also, it reduces the complexity of the settings page by refactoring it.
Refactored and improved prop-types.
Refactored code in all files related to this feature to improve it readability.

🖥 UI Demonstration: 

🛠 DEV UI BELOW  ⬇️⬇️⬇️  (Chrome browser)

🎥 Small loom-demo (link):  https://www.loom.com/share/58bf322b70a84bbdb49814dc5395f34c

🏭 PRODUCTION UI BELOW  ⬇️⬇️⬇️ (Electron App, Mac OS)

Снимок экрана 2021-03-18 в 15 06 36
Снимок экрана 2021-03-18 в 15 08 28

### CODE REVIEW

**Review by alexstotsky — APPROVED** [review]
lgtm

File: src/pages/Settings/Settings.js [hunk 1]
```diff
@@ -1,66 +1,58 @@
...
-        dms, ga,
-      } = this.props
-      this.setState(() => ({
-        dms, ga,
-      }))
+      apiKey: '',
+      apiSecret: '',
+      isDevEnv: devEnv(),
+      currentExchange: activeExchange,
+      AUTOLOGIN_STATE: getAutoLoginState(),
```
Comment (bhoomij) [inline src/pages/Settings/Settings.js:48]: same here, could use isDevEnv/AUTOLOGIN_STATE without state vars

**Review by bhoomij — CHANGES_REQUESTED** [review]
great job!! please see comment for state vars

**Review by bhoomij — APPROVED** [review]

**Comment by robertkowalski** [comment]
Screenshot 2021-03-22 at 11 29 07

how do i use that? can't select the checkbox

is that visible to users? that would mean in prod users will use dev features? shouldn't it be hidden for users in prod?

another problem that comes with it: users will use it at some point out of laziness, and all the security is gone

**Pull request merged**